README.md

```diff
@@ -1,130 +1,191 @@
-```javascript
-// note: state and forcePrompt are optional
-smartcar.addClickHandler({id, state, forcePrompt});
-```
+Once initialized there are three instance methods for setting up the OAuth flow:
+
+- `openDialog()` - to open the Smartcar OAuth dialog directly
+- `addClickHandler({id})` - to add a click handler to an HTML element that will open the Smartcar OAuth dialog
+<!-- TODO: what is this section trying to communicate? is the 'useful in a single page app...' portion still relevant? -->
+- `generateLink()` - to generate the bare Smartcar OAuth URL (useful in single page applications when redirecting to Smartcar's OAuth flow in a new tab)
+
+See the [reference below](TODO: link to appropriate section) for detailed signature information.
+
+<!-- organization of this whole section a little awkward with this ending another example of the awkwardness -->
+You may optionally embed `redirect.js` in the page that is served by your redirect URI. Upon loading, this file will trigger invocation of the `onComplete` callback specified above by posting to the client and will close out the redirect window:
 
-You may embed `callback.js` in the page that is served by your redirectUri. Upon loading, this file will invoke the `onComplete` callback specified above and close the window:
 ```html
 <script src="https://cdn.smartcar.com/javascript-sdk/callback-1.0.0.js"></script>
 ```
 
-To retrieve the Smartcar OAuth URL (useful in single page applications when redirecting to Smartcar's OAuth flow in a new tab):
-```javascript
-// note: state and forcePrompt are optional
-smartcar.generateLink({state, forcePrompt});
-```
-
 ## Installation
+
 You can import this SDK into your application from Smartcar's CDN:
 
 ### sdk.js
+
 ```html
 <script src="https://cdn.smartcar.com/javascript-sdk/sdk-1.0.0.js"></script>
 ```
 
-### callback.js
+### redirect.js
+
 ```html
 <script src="https://cdn.smartcar.com/javascript-sdk/callback-1.0.0.js"></script>
 ```
 
 ## Configuration
+
 ### `new Smartcar({options})`
+
 #### Options:
 | Parameter       | Type | Description   |
 |:--------------- |:---|:------------- |
-| `clientId`      | String |Application clientId obtained from [Smartcar Developer Portal](https://developer.smartcar.com). If you do not have access to the dashboard, please [request access](https://smartcar.com/subscribe). |
-| `redirectUri`   | String |**Required** RedirectURI set in [application settings](https://developer.smartcar.com/apps). Given URL must match URL in application settings. |
+| `clientId`      | String |Application clientId obtained from [Smartcar Developer Portal](https://dashboard.smartcar.com). |
+| `redirectUri`   | String |**Required** Redirect URI set in [application settings](https://dashboard.smartcar.com/apps). Given URL must match URL in application settings. To use Smartcar hosting this URI must be of the form `https://javascript-sdk.smartcar.com/redirect?app_origin=${your client URI here}` |
```

Comment:
maybe add another row documenting what the "client URI" means again

Reply:
Do you mean a row in this table? "client URI" isn't an option so that seems odd to me. Or do you mean like another sentence in the `redirectUri` row explaining what "client URI" is?

Reply:
you're right, it would be weird for it to be another row. maybe either another sentence in here, or somewhere else in the README explaining what it does